lib/storage.js

```diff
@@ -69,6 +72,13 @@ class Storage {
     this.sources.forEach((source) => {
       merge(source.properties, properties);
     });
+
+    const diff = deepDiff(this.properties, properties);
+
+    if (diff) {
+      Log.info(`Calculated property diff`, {diff});
+    }
+
     this.properties = properties;
```

Comment:
You're right but I'd like to understand the diffs before we confirm that properties should only be updated if there is a diff as calculated by this lib. I'm adding more diff logging so we can get some insight into the current behavior.